src/Service.Tests/SqlTests/RestApiTests/RestBodyNonStrictModeTests/RestBodyNonStrictModeTests.cs

```diff
@@ -0,0 +1,180 @@
+            requestBody = @"
+            {
+               ""categoryName"":""SciFi"",
+               ""piecesAvailable"":""10"",
+               ""piecesRequired"":""5"",
+               ""piecied"": 5,
+               ""categoryid"": 4
+            }";
+
+            await SetupAndRunRestApiTest(
+                primaryKeyRoute: "categoryid/2/pieceid/1",
+                queryString: null,
+                entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                sqlQuery: GetQuery("PutOneWithPKFieldsInRequestBody"),
+                operationType: EntityActionOperation.Upsert,
+                requestBody: requestBody,
+                expectedStatusCode: HttpStatusCode.OK
+                );
+
+            // Validate that when fields which do not map to any column (here 'non_existing_field')
+            // in the underlying table are included in the request body for PUT operation,
+            // they are just ignored when we do not operate in strict mode for REST.
+            requestBody = @"
+            {
+               ""categoryName"":""SciFi"",
+               ""piecesAvailable"":""10"",
+               ""piecesRequired"":""5"",
+               ""non_existing_field"": ""5""
+            }";
+
+            await SetupAndRunRestApiTest(
+                primaryKeyRoute: "categoryid/2/pieceid/1",
+                queryString: null,
+                entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                sqlQuery: GetQuery("PutOneWithNonExistingFieldInRequestBody"),
+                operationType: EntityActionOperation.Upsert,
+                requestBody: requestBody,
+                expectedStatusCode: HttpStatusCode.OK
+                );
+
+            requestBody = @"
+            {
+                ""piecesAvailable"": null,
+                ""piecied"": 5,
+                ""categoryid"": 4
+            }";
+
+            // Validate that when PK fields (here 'categoryid' and 'pieceid') are included in the request body for PATCH operation,
+            // they are just ignored when we do not operate in strict mode for REST. The values of the fields from the URL is used.
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "categoryid/1/pieceid/1",
+                    queryString: null,
+                    entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                    sqlQuery: GetQuery("PatchOneWithPKFieldsInRequestBody"),
+                    operationType: EntityActionOperation.UpsertIncremental,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.OK
+                );
+
+            requestBody = @"
+            {
+                ""piecesAvailable"": null,
+                ""non_existing_field"": ""5""
+            }";
+
+            // Validate that when fields which do not map to any column (here 'non_existing_field')
```

Comment:
Do we already have tests for erroring out when there are extraneous fields in the default mode?

Reply:
There was one in insert tests. Added more to all the different mutations.